src/lib/php/Dao/ClearingDao.php

```diff
@@ -618,62 +618,39 @@ public function markDecisionAsWip($uploadTreeId, $userId, $groupId)
     $this->dbManager->freeResult($res);
   }
 
-  public function isDecisionWip($uploadTreeId, $groupId)
-  {
-    $sql = "SELECT decision_type FROM clearing_decision WHERE uploadtree_fk=$1 AND group_fk = $2 ORDER BY date_added DESC LIMIT 1";
-    $latestDec = $this->dbManager->getSingleRow($sql,
-                 array($uploadTreeId, $groupId), $sqlLog = __METHOD__);
-    if ($latestDec === false) {
-      return false;
-    }
-    return ($latestDec['decision_type'] == DecisionTypes::WIP);
-  }
-
-  public function isDecisionTBD($uploadTreeId, $groupId)
-  {
-    $sql = "SELECT decision_type FROM clearing_decision WHERE uploadtree_fk=$1 AND group_fk = $2 ORDER BY date_added DESC LIMIT 1";
-    $latestDec = $this->dbManager->getSingleRow($sql,
-                 array($uploadTreeId, $groupId), $sqlLog = __METHOD__);
-    if ($latestDec === false) {
-      return false;
-    }
-    return ($latestDec['decision_type'] == DecisionTypes::TO_BE_DISCUSSED);
-  }
-
-  public function isDecisionDNU($uploadTreeId, $groupId)
+  /**
+   * @param int $uploadTreeId
+   * @param int $groupId
+   * @param Char $decisionType
+   */
+  public function isDecisionCheck($uploadTreeId, $groupId, $decisionType)
   {
-    $sql = "SELECT decision_type FROM clearing_decision
-              WHERE uploadtree_fk=$1 AND group_fk = $2
-            ORDER BY clearing_decision_pk DESC LIMIT 1";
-    $latestDec = $this->dbManager->getSingleRow($sql,
-                 array($uploadTreeId, $groupId), $sqlLog = __METHOD__);
-    if ($latestDec === false) {
-      return false;
+    $isDecisionType = "";
+    $columns = "decision_type";
+    if ($decisionType === 'WIP') {
+      $isDecisionType = DecisionTypes::WIP;
+    } else if ($decisionType === 'TBD') {
+      $isDecisionType = DecisionTypes::TO_BE_DISCUSSED;
+    } else if ($decisionType === 'DNU') {
+      $isDecisionType = DecisionTypes::DO_NOT_USE;
+    } else if ($decisionType === 'IRR') {
+      $isDecisionType = DecisionTypes::IRRELEVANT;
+    } else {
+      $columns = "decision_type, scope";
     }
```

Comment:
If using class constants like `DecisionTypes::IRRELEVANT` while calling the function, this condition can simply be reduced to
```php
if (!in_array($decisionType, [DecisionTypes::WIP, DecisionTypes::TO_BE_DISCUSSED, DecisionTypes::DO_NOT_USE, DecisionTypes::IRRELEVANT]) {
  $columns = "decision_type, scope";
  $decisionType = "";
}
```